File: app/src/main/java/com/developer/valyutaapp/ui/splash/SplashActivity.java
package com.developer.valyutaapp.ui.splash;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.developer.valyutaapp.R;
import com.developer.valyutaapp.ui.main.MainActivity;

import butterknife.BindView;

public class SplashActivity extends AppCompatActivity {


    @Nullable
    @BindView(R.id.tvTitle)
    TextView tvTitle;

    private Intent intent;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
//        tvTitle.setText(R.string.splash);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                intent = new Intent(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        }, 3 * 1000);
    }
}

File: app/src/main/java/com/developer/valyutaapp/ui/splash/SplashActivity.kt
package com.developer.valyutaapp.ui.splash

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.developer.valyutaapp.ui.MainActivity
import com.developer.valyutaapp.utils.Utils.setStatusBar

@SuppressLint("CustomSplashScreen")
class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStatusBar(window)
        splashAndroid12()
        postDelay()
    }

    private fun splashAndroid12() {
        val content = findViewById<View>(android.R.id.content)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            content.viewTreeObserver.addOnDrawListener { false }
        }
    }

    private fun postDelay() {
        Handler(Looper.getMainLooper()).postDelayed({ callMain() }, 1000)
    }

    private fun callMain() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
}